com.microsoft.mrtk.uxcomponents/Editor/CreateElementMenus.cs

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEditor.UI;
+using UnityEngine.EventSystems;
+using TMPro;
+using System.Reflection;
+using System.Linq;
+
+namespace Microsoft.MixedReality.Toolkit.Editor
+{
+    /// <summary>
+    /// Add-GameObject-menus for creating common MRTK UX components. Most of these
+    /// spawn prefabs, but some are just ad-hoc objects created in code (like panels,
+    /// backplates, etc).
+    /// </summary>
+    static internal class CreateElementMenus
+    {
+        // EmptyButton.prefab
+        // A completely empty button, with no icon, text, label, etc.
+        private static readonly string EmptyButtonPath = AssetDatabase.GUIDToAssetPath("b85e005d231192249b7077b40a4d4e45");
+
+        // ActionButton.prefab
+        // The basic building block button; contains an icon, text, and label.
+        private static readonly string ActionButtonPath = AssetDatabase.GUIDToAssetPath("c6b351a67ceb69140b199996bbbea156");
+
+        // Reflection into internal UGUI editor utilities.
+        private static System.Reflection.MethodInfo PlaceUIElementRoot = null;
```

Comment:
is there a reason you explicitly added System.Reflection here?

Reply:
ok, i see. Unity has a MethodInfo struct defined in the editor's profiling namespace